dashboard/app/controllers/pd/facilitator_program_registration_controller.rb

```diff
@@ -29,11 +29,14 @@ def new
     end
 
     @script_data = {
-      options: Pd::FacilitatorProgramRegistration.options.camelize_keys.to_json,
-      attendanceDates: dates.camelize_keys.to_json,
-      teachercon: @teachercon,
-      course: Pd::FacilitatorProgramRegistration.course(current_user, @teachercon).to_json,
-      teacherconLocation: Pd::FacilitatorProgramRegistration::LOCATIONS[@teachercon - 1].to_json
+      props: {
+        options: Pd::FacilitatorProgramRegistration.options.camelize_keys,
+        requiredFields: Pd::FacilitatorProgramRegistration.camelize_required_fields,
+        attendanceDates: dates.camelize_keys,
+        teachercon: @teachercon,
+        course: Pd::FacilitatorProgramRegistration.course(current_user, @teachercon),
+        teacherconLocation: Pd::FacilitatorProgramRegistration::LOCATIONS[@teachercon - 1]
+      }.to_json
```

Comment:
Out of curiosity, why wrap everything in a nested `props` object? Is that just to future proof in case we want other script_data that is not props for the top level component?